cfme/tests/infrastructure/test_providers.py

```diff
@@ -520,14 +521,19 @@ def test_compare_provider_templates(appliance, setup_provider_min_templates, pro
     Bugzilla:
         1746449
     """
-    templateList = []
-    my_slice = slice(0, min_templates, None)
-    view = navigate_to(provider, 'ProviderTemplates')
-    for t in view.entities.get_all(slice=my_slice):
-        t.ensure_checked()
-        templateList.append(t.name)
-    view.toolbar.configuration.item_select('Compare Selected Templates', handle_alert=True)
-    compare_templates_view = provider.create_view(TemplatesCompareView)
-    assert compare_templates_view.is_displayed
-    assert compare_templates_view.verify_checked_items_compared(templateList,
-                                                                compare_templates_view)
+    if templates_collection == 'provider':
+        provider.collections.templates = InfraTemplateCollection(provider,
```

Comment:
These collections should not be directly instantiated, as with hosts it should be getting populated automatically.

Reply:
I am having a bit of trouble getting this to work without directly instantiating the collection for the provider case. I have it working for appliances. I need to determine 1) the correct way to get the collection to use in the call. 2) Any updates to the navigation that I need to do to support this. I believe I may be going in circles here. there are several "patterns" I am seeing in the framework, but I am likely using them incorrectly and don't really understand the underlying logic for many of these?

Reply:
No longer directly instantiated.